### PR TITLE
CI-windows: make Bazel operate outside of C:

### PR DESCRIPTION
**Pull request by hevrard**
This can hopefully reduce flakes.

Bug: b/182387034

### CODE REVIEW

**Comment by hevrard** [comment]
Maybe it would be even better to just set HOME to BUILD_ROOT at the start, such that all commands see the same HOME (rather than backing up HOME and restoring it). Let me know your thoughts.

**Comment by johnplate** [comment]
> Maybe it would be even better to just set HOME to BUILD_ROOT at the start, such that all commands see the same HOME (rather than backing up HOME and restoring it). Let me know your thoughts.

I would not change HOME at all, might have bad side effects. On my Windows machine I use a .bazelrc file in my user directory, only containing the line `startup --output_user_root=D:/scratch/bazel`, which works fine. It would be even easier to add this option to both bazel commands instead of creating that file.

**Review by pmuetschard — COMMENTED** [review]
1) I'm not convinced that this is the root cause of the flakes. So, I would remove that comment.
2) I agree with John, don't touch `%HOME%`, but pass the output root flag to each bazel invocation, instead.

**Comment by hevrard** [comment]
Thanks both for the comments, I've updated the change accordingly, PTAL

**Review by johnplate — APPROVED** [review]
LGTM, but wait for the Windows CI to succeed and check if bazel actually uses the given directory.

**Comment by johnplate** [comment]
> LGTM, but wait for the Windows CI to succeed and check if bazel actually uses the given directory.

I checked it already, works fine, ready to submit.

**Comment by pmuetschard** [comment]
make sure you rebase on top of master and also add the flag to the smoketest invocation.

**Comment by johnplate** [comment]
> make sure you rebase on top of master and also add the flag to the smoketest invocation.

I'm confused, why would the smoketest invocation need the flag?

**Comment by hevrard** [comment]
@johnplate Pascal recently refactored the way smoketests is run, so after rebasing, this also needs the flag.

**Comment by johnplate** [comment]
> @johnplate Pascal recently refactored the way smoketests is run, so after rebasing, this also needs the flag.

Thanks, got it. New push still LGTM

**Comment by hevrard** [comment]
Rebased on master to have CI-windows download errors fixed.

File: kokoro/windows/build.bat [hunk 1]
```diff
@@ -104,22 +104,36 @@ if "%KOKORO_GITHUB_COMMIT%." == "." (
   set BUILD_SHA=%DEV_PREFIX%%KOKORO_GITHUB_COMMIT%
 )
 
-%BUILD_ROOT%\bazel build -c opt --config symbols ^
+REM Use T:\src for Bazel output, where files are less likely to be
+REM locked by system checks.
+set BAZEL_OUTPUT_USER_ROOT="T:\src\bazel_user_root"
```
Comment (pmuetschard) [inline kokoro/windows/build.bat:109]: Use `%BUILD_ROOT%` and name it something short. `out` is already used so why not `%BUILD_ROOT%\build`?

Comment (hevrard) [reply kokoro/windows/build.bat]: done

**Review by pmuetschard — APPROVED** [review]

**Pull request merged**